data_transfer/devices/dreem.py

```diff
@@ -88,53 +95,37 @@ def download_metadata(self) -> None:
                 )
             )
 
-            # Reformat to mirror UCAM/DMP.
-            patient_id = patient_id.replace("-", "") if patient_id else patient_id
-
-            if patient_id and (ucam_entry := ucam.get_record(patient_id)):
-                known += 1
-                dreem_devices = [
-                    d for d in ucam_entry.devices if self.device_type in d.device_id
-                ]
-
-                # Best-case: only one device was worn and UCAM knows it
-                if len(dreem_devices) == 1:
-                    device_record = dreem_devices[0]
-                # Edge-case: multiple dreem headbands used, e.g., if one broke.
-                elif len(dreem_devices) > 1:
-                    # Determine usage based on weartime as it exists in UCAM
-                    device_record = ucam.record_by_wear_period_in_list(
-                        dreem_devices, recording.start, recording.end
-                    )
-                # Edge-case: device not logged with patient in UCAM
-                else:
-                    print(f"Metadata cannot be determined for:\n    {recording}")
-                    continue
-            else:
+            device_id = self.__device_id_from_ucam(
+                patient_id, recording.start, recording.end
+            ) or inventory.device_id_by_serial(device_serial)
+
+            if not patient_id or not device_id:
+                log.debug(f"Unknown Device:\n   {recording}")
+                log.debug(f"Patient ({patient_id}) or Device ID ({device_id}) missing.")
                 unknown += 1
-                print(f"Metadata cannot be determined for:\n    {recording}")
-                continue
+                continue  # Skip record
+            known += 1
 
             record = Record(
-                filename=recording.id,
+                hash=recording.id,
```

Comment:
Can you please use the [hasher](https://github.com/ideafast/middleware-services/blob/17a9674bf803deb619a5e507a11549af72cc8706/data_transfer/utils/__init__.py#L151-L157) to set this value? 

In the BTF branch, I calculate this hash for each recording and add it to the metadata. I'd suggest to do the same for this DRM branch. Effectively, in the DRM lib before [this line](https://github.com/ideafast/middleware-services/blob/17a9674bf803deb619a5e507a11549af72cc8706/data_transfer/lib/dreem.py#L57), add :
```python
result["IDEAFAST"]["hash"] = uid_to_hash( result["id"], DeviceType.DRM )
```

Then this line here, should become:
```python
hash=item["IDEAFAST"]["hash"]
```
which is identical to the BTF implementation

Reply:
How come we add it to the metadata? I could certainly change this to use `uid_to_hash` and update the set difference above, but it's unclear why we would send this middleware-specific hash to the DMP?

Reply:
Resolved via adding `__unknown_records` method